build-system/common/git.js

```diff
@@ -169,12 +169,36 @@ function gitCommitterEmail() {
 }
 
 /**
- * Returns the timestamp of the latest commit on the local branch.
- * @return {number}
+ * Returns list of commit SHAs and their cherry-pick status from master.
+ *
+ * `git cherry <branch>` returns a list of commit SHAs. While the exact
+ * mechanism is too complicated for this comment (run `git help cherry` for a
+ * full explanation), the gist of it is that commits that were cherry-picked
+ * from <branch> are prefixed with '- ', and those that were not are prefixed
+ * with '+ '.
+ *
+ * @return {!Array<{sha: string, isCherryPick: boolean}>}
+ */
+function gitCherryMaster() {
+  return getStdout('git cherry master')
```

Comment:
A use case I just thought of: Today, it isn't necessary for developers to keep their local `master` branch in sync with `upstream/master`. For example: 

- A developer has a `developer/amphtml` fork that is a couple of months old
- They create a local branch `foo`
- Sync `foo` to `upstream/master`
- Add a bunch of commits to it
- Meanwhile, the local `master` branch is still two months old

Will this result in the new versioning logic detecting hundreds of "cherry-picked" commits?

Reply:
Answering my own question: This isn't a problem because the additional commits between the outdated local `master` and `HEAD` are counted as new commits, not cherry-picked commits. Carry on!

<img width="1199" alt="Screen Shot 2020-04-22 at 2 46 10 PM" src="https://user-images.githubusercontent.com/26553114/80021379-6b6e8a00-84a8-11ea-9a39-cafbd3c99d26.png">

Reply:
This is exactly why I stop counting once I get to the first not-cherry-pick commit :)